src/main/java/net/minecraftforge/common/util/MutableHashedLinkedList.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+package net.minecraftforge.common.util;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.apache.commons.lang3.mutable.MutableObject;
+import org.jetbrains.annotations.Nullable;
+
+import it.unimi.dsi.fastutil.Hash.Strategy;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenCustomHashMap;
+
+public class MutableHashedLinkedList<K, V> implements Iterable<Map.Entry<K, V>>
+{
+    private final Map<K, Entry> entries;
+    private final MergeFunction<K, V> merge;
+    private final MutableObject<Entry> head = new MutableObject<>();
+    private final MutableObject<Entry> last = new MutableObject<>();
+
+    public MutableHashedLinkedList(Strategy<? super K> strategy) {
```

Comment:
Formatting: braces on newlines.

Please also fix the formatting in the rest of the class.